runtime/src/main/java/org/corfudb/runtime/object/transactions/WriteSetSMRStream.java

```diff
@@ -167,7 +171,7 @@ public void gc(long trimMark) {
     @Override
     public List<SMREntry> current() {
         if (Address.nonAddress(writePos)) {
-            return null;
+            return new ArrayList<>();
```

Comment:
This is inconsistent. You changed this method to never return null, but you didn't change the same overridden method in StreamViewAdapter. Also in VersionLockedObject, the consumer of this method is still checking if the return value is null.

Reply:
Good catch, thanks. So, I should finish the cleanup.
Will provide another pr soon

Reply:
https://github.com/CorfuDB/CorfuDB/pull/2033